### PR TITLE
fix next hero focusing the first hero if used when the focus is on a castle

### PR DESCRIPTION
**Pull request by eos428**
Solves #2202

### CODE REVIEW

File: src/fheroes2/gui/interface_events.cpp [hunk 1]
```diff
@@ -138,7 +138,13 @@ void Interface::Basic::EventNextHero( void )
         } while ( it != currentHero );
     }
     else {
-        ResetFocus( GameFocus::HEROES );
+        const int heroesCount = myHeroes.size();
+        for ( int i = 0; i < heroesCount; ++i ) {
```
Comment (ihhub) [inline src/fheroes2/gui/interface_events.cpp:142]: Please use `size_t` instead.

**Comment by ihhub** [comment]
Hi @eos428 I'll review all your pull requests soon. I'm a bit busy these days :(

**Comment by eos428** [comment]
No worries, take your time.

**Comment by ihhub** [comment]
@eos428 thank you very much for your valuable contribution to the project!

**Pull request merged**